packages/ra-core/src/dataProvider/useQuery.ts

```diff
@@ -143,6 +144,7 @@ export interface Query {
 
 export interface QueryOptions {
     action?: string;
+    enabled?: boolean;
```

Comment:
You should add options.enabled to the dependency list of the useEffect, as you did in usequeryWithStore

Reply:
Having `options.enabled` in the useEffect there is actually a left-over that I forgot to remove when removing the early return in the hook. As the logic is now handled in `useDataProvider` and this option is part of the `requestSignature`, I don't think this is  needed.